ctapipe/io/hdf5merger.py

```diff
@@ -0,0 +1,467 @@
+            )
+
+        self.required_nodes = set()
+        if self.overwrite:
+            appending = False
+
+        self.h5file = tables.open_file(
+            self.output_path,
+            mode="w" if self.overwrite else "a",
+            filters=DEFAULT_FILTERS,
+        )
+        Provenance().add_output_file(str(self.output_path))
+
+        self.data_model_version = None
+        self.subarray = None
+        self.meta = None
+        # output file existed, so read subarray and data model version to make sure
+        # any file given matches what we already have
+        if appending:
+            try:
+                self.meta = metadata.Reference.from_dict(
+                    metadata.read_metadata(self.h5file)
+                )
+            except Exception:
+                raise CannotMerge(
+                    f"CTA Rerence meta not found in existing output file: {self.output_path}"
+                )
+
+            self.data_model_version = self.meta.product.data_model_version
+
+            # focal length choice doesn't matter here, set to equivalent so we don't get
+            # an error if only the effective focal length is available in the file
+            self.subarray = SubarrayDescription.from_hdf(
+                self.h5file, focal_length_choice=FocalLengthKind.EQUIVALENT
+            )
+            self.required_nodes = _get_required_nodes(self.h5file)
+
+    def __call__(self, other: Union[str, Path, tables.File]):
+        """
+        Append file ``other`` to the ouput file
+        """
+        exit_stack = ExitStack()
+        if not isinstance(other, tables.File):
+            other = exit_stack.enter_context(tables.open_file(other, mode="r"))
+
+        with exit_stack:
+            self._check_can_merge(other)
+
+            Provenance().add_input_file(other.filename)
+            try:
+                self._append(other)
+            finally:
+                self._update_meta()
+
+    def _update_meta(self):
+        # update creation date and id
+        time = Time.now()
+        id_ = str(uuid.uuid4())
+        self.meta.product.id_ = id_
+        self.meta.product.creation_time = time
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", tables.NaturalNameWarning)
+            self.h5file.root._v_attrs["CTA PRODUCT CREATION TIME"] = time.iso
+            self.h5file.root._v_attrs["CTA PRODUCT ID"] = id_
+        self.h5file.flush()
+
+    def _check_can_merge(self, other):
+        try:
+            other_meta = metadata.Reference.from_dict(metadata.read_metadata(other))
+        except Exception:
+            raise CannotMerge(
+                f"CTA Rerence meta not found in input file: {other.filename}"
+            )
+
+        if self.meta is None:
+            self.meta = other_meta
+            self.data_model_version = self.meta.product.data_model_version
+            metadata.write_to_hdf5(self.meta.to_dict(), self.h5file)
```

Comment:
It did come as a surprise here, that this function actually writes the metadata and not just checks

Reply:
yes, true, will change.